### ISSUE TITLE
Field filter doesn't work

### ISSUE THREAD

**Issue by qnkhuat** [metabase/metabase#37630]
### Describe the bug

Field filter doesn't work

### To Reproduce

1. Create a native question `select * from products where {{category}}` and configure category to be a field filter and connect to Products.category
2. Save the question and refresh the page (I'm not sure if this is necessary or not but sometimes the bug doesn't surface without it)
3. Try to select a value for the filter, see the error

### Logs

``` 
converted pMBQL query should validate against the pMBQL schema
expected: (not (me/humanize (mr/explain :metabase.lib.schema/query pMBQL)))
  actual: (not (not {:stages [{:template-tags {"category" {:options ["invalid type"]}}}]}))

FAIL in (test-mlv2-conversion) (test_mlv2.clj:63)

legacy query =
{:constraints {:max-results 10000, :max-results-bare-rows 2000},
 :type :native,
 :middleware {:js-int-to-string? true, :ignore-cached-results? false, :process-viz-settings? false},
 :native
 {:template-tags
  {"category"
   {:type :dimension,
    :name "category",
    :id "ce36b07d-dfbb-4db6-b7f3-e7a46e3818bd",
    :display-name "Category",
    :default nil,
    :dimension [:field 201027 {:base-type :type/Text}],
    :widget-type :string/=,
    :options nil}},
  :query "select * from products where {{category}}"},
 :info {:executed-by 3, :context :question, :card-id 77, :card-name "bugggg"},
 :database 7,
 :parameters
 [{:id "ce36b07d-dfbb-4db6-b7f3-e7a46e3818bd",
   :type :string/=,
   :value ["Doohickey"],
   :target [:dimension [:template-tag "category"]]}],
 :async? true}


pMBQL =
{:constraints {:max-results 10000, :max-results-bare-rows 2000},
 :lib/type :mbql/query,
 :stages
 [{:lib/type :mbql.stage/native,
   :template-tags
   {"category"
    {:type :dimension,
     :name "category",
     :id "ce36b07d-dfbb-4db6-b7f3-e7a46e3818bd",
     :display-name "Category",
     :default nil,
     :dimension [:field {:base-type :type/Text, :lib/uuid "ace670c8-88c0-49e6-a80a-038730ad3227"} 201027],
     :widget-type :string/=,
     :options nil}},
   :native "select * from products where {{category}}"}],
 :middleware {:js-int-to-string? true, :ignore-cached-results? false, :process-viz-settings? false},
 :lib.convert/converted? true,
 :info {:executed-by 3, :context :question, :card-id 77, :card-name "bugggg"},
 :database 7,
 :parameters
 [{:id "ce36b07d-dfbb-4db6-b7f3-e7a46e3818bd",
   :type :string/=,
   :value ["Doohickey"],
   :target [:dimension [:template-tag "category"]]}],
 :async? true}
 ```

### Information about your Metabase installation

```JSON
master, v48 is fine
```

### Severity

P2

**Comment by qnkhuat** [comment]
might be the root cause of https://github.com/metabase/metabase/issues/37623

**Comment by bshepherdson** [comment]
It wasn't the cause of that bug, but I fixed it in passing.

#37623 is a dupe of #38012, which was fixed in #38070. I saw and suspected this error while debugging that, but the error was a nuisance and not the real issue. In any case, #38070 fixes this error in passing, so see if you can still reproduce this on master or 48.4.

**Comment by paoliniluis** [comment]
Fixed
